# NYC Airbnb Data Overview

## Objective
Explore and understand the structure, size, and quality of the NYC Airbnb dataset
before performing data cleaning and analysis.


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [3]:
data = pd.read_csv('../data/raw/Airbnb_Open_Data.csv', low_memory = False)

In [4]:
data.head()

id                                              NAME      host id  \
0  1001254                Clean & quiet apt home by the park  80014485718   
1  1002102                             Skylit Midtown Castle  52335172823   
2  1002403               THE VILLAGE OF HARLEM....NEW YORK !  78829239556   
3  1002755                                               NaN  85098326012   
4  1003689  Entire Apt: Spacious Studio/Loft by central park  92037596077   

  host_identity_verified host name neighbourhood group neighbourhood  \
0            unconfirmed  Madaline            Brooklyn    Kensington   
1               verified     Jenna           Manhattan       Midtown   
2                    NaN     Elise           Manhattan        Harlem   
3            unconfirmed     Garry            Brooklyn  Clinton Hill   
4               verified    Lyndon           Manhattan   East Harlem   

        lat      long        country  ... service fee minimum nights  \
0  40.64749 -73.97237  United States  ...       $193            10.0   
1  40.75362 -73.98377  United States  ...        $28            30.0   
2  40.80902 -73.94190  United States  ...       $124             3.0   
3  40.68514 -73.95976  United States  ...        $74            30.0   
4  40.79851 -73.94399  United States  ...        $41            10.0   

  number of reviews last review  reviews per month review rate number  \
0               9.0  10/19/2021               0.21                4.0   
1              45.0   5/21/2022               0.38                4.0   
2               0.0         NaN                NaN                5.0   
3             270.0    7/5/2019               4.64                4.0   
4               9.0  11/19/2018               0.10                3.0   

  calculated host listings count  availability 365  \
0                            6.0             286.0   
1                            2.0             228.0   
2                            1.0             352.0   
3                            1.0             322.0   
4                            1.0             289.0   

                                         house_rules license  
0  Clean up and treat the home the way you'd like...     NaN  
1  Pet friendly but please confirm with me if the...     NaN  
2  I encourage you to use my kitchen, cooking and...     NaN  
3                                                NaN     NaN  
4  Please no smoking in the house, porch or on th...     NaN  

[5 rows x 26 columns]

In [5]:
data.tail()

id                              NAME      host id  \
102594  6092437        Spare room in Williamsburg  12312296767   
102595  6092990     Best Location near Columbia U  77864383453   
102596  6093542    Comfy, bright room in Brooklyn  69050334417   
102597  6094094  Big Studio-One Stop from Midtown  11160591270   
102598  6094647              585 sf Luxury Studio  68170633372   

       host_identity_verified    host name neighbourhood group  \
102594               verified         Krik            Brooklyn   
102595            unconfirmed        Mifan           Manhattan   
102596            unconfirmed        Megan            Brooklyn   
102597            unconfirmed  Christopher              Queens   
102598            unconfirmed      Rebecca           Manhattan   

              neighbourhood       lat      long        country  ...  \
102594         Williamsburg  40.70862 -73.94651  United States  ...   
102595  Morningside Heights  40.80460 -73.96545  United States  ...   
102596           Park Slope  40.67505 -73.98045  United States  ...   
102597     Long Island City  40.74989 -73.93777  United States  ...   
102598      Upper West Side  40.76807 -73.98342  United States  ...   

       service fee minimum nights number of reviews last review  \
102594       $169             1.0               0.0         NaN   
102595       $167             1.0               1.0    7/6/2015   
102596       $198             3.0               0.0         NaN   
102597       $109             2.0               5.0  10/11/2015   
102598       $206             1.0               0.0         NaN   

        reviews per month review rate number calculated host listings count  \
102594                NaN                3.0                            1.0   
102595               0.02                2.0                            2.0   
102596                NaN                5.0                            1.0   
102597               0.10                3.0                            1.0   
102598                NaN                3.0                            1.0   

        availability 365                                        house_rules  \
102594             227.0  No Smoking No Parties or Events of any kind Pl...   
102595             395.0  House rules: Guests agree to the following ter...   
102596             342.0                                                NaN   
102597             386.0                                                NaN   
102598              69.0                                                NaN   

       license  
102594     NaN  
102595     NaN  
102596     NaN  
102597     NaN  
102598     NaN  

[5 rows x 26 columns]

In [11]:
data.shape

(102599, 26)

- About 102599 rows are present in the dataset and 26 columns

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102599 entries, 0 to 102598
Data columns (total 26 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   id                              102599 non-null  int64  
 1   NAME                            102349 non-null  object 
 2   host id                         102599 non-null  int64  
 3   host_identity_verified          102310 non-null  object 
 4   host name                       102193 non-null  object 
 5   neighbourhood group             102570 non-null  object 
 6   neighbourhood                   102583 non-null  object 
 7   lat                             102591 non-null  float64
 8   long                            102591 non-null  float64
 9   country                         102067 non-null  object 
 10  country code                    102468 non-null  object 
 11  instant_bookable                102494 non-null  object 
 12  cancellation_pol

In [7]:
data.isna().sum()

id                                     0
NAME                                 250
host id                                0
host_identity_verified               289
host name                            406
neighbourhood group                   29
neighbourhood                         16
lat                                    8
long                                   8
country                              532
country code                         131
instant_bookable                     105
cancellation_policy                   76
room type                              0
Construction year                    214
price                                247
service fee                          273
minimum nights                       409
number of reviews                    183
last review                        15893
reviews per month                  15879
review rate number                   326
calculated host listings count       319
availability 365                     448
house_rules     

In [8]:
data.isna().sum().sum()

np.int64(190769)

- Names of the columns are not consistent, used different formatting
- Total of 190769 values are missing from the dataset


In [13]:
(data.isna().sum() / data.shape[0] * 100).sort_values(ascending=False)

license                           99.998051
house_rules                       50.810437
last review                       15.490404
reviews per month                 15.476759
country                            0.518524
availability 365                   0.436651
minimum nights                     0.398639
host name                          0.395715
review rate number                 0.317742
calculated host listings count     0.310919
host_identity_verified             0.281679
service fee                        0.266084
NAME                               0.243667
price                              0.240743
Construction year                  0.208579
number of reviews                  0.178364
country code                       0.127682
instant_bookable                   0.102340
cancellation_policy                0.074075
neighbourhood group                0.028265
neighbourhood                      0.015595
long                               0.007797
lat                             

- `license` has almost complete missingness at 99.998%, meaning it is essentially unusable for analysis.
- `house_rules` has a high missing rate at 50.81%, so half of the entries lack this information.
- `last review` and `reviews per month` both have around 15% missing values, which is moderate and may require imputation or careful handling.
- Most other columns have very low missingness, typically below 1%, including country, availability 365, minimum nights, and many others.

In [ ]:
data.head()

id                                              NAME      host id  \
0  1001254                Clean & quiet apt home by the park  80014485718   
1  1002102                             Skylit Midtown Castle  52335172823   
2  1002403               THE VILLAGE OF HARLEM....NEW YORK !  78829239556   
3  1002755                                               NaN  85098326012   
4  1003689  Entire Apt: Spacious Studio/Loft by central park  92037596077   

  host_identity_verified host name neighbourhood group neighbourhood  \
0            unconfirmed  Madaline            Brooklyn    Kensington   
1               verified     Jenna           Manhattan       Midtown   
2                    NaN     Elise           Manhattan        Harlem   
3            unconfirmed     Garry            Brooklyn  Clinton Hill   
4               verified    Lyndon           Manhattan   East Harlem   

        lat      long        country  ... service fee minimum nights  \
0  40.64749 -73.97237  United States  ...       $193            10.0   
1  40.75362 -73.98377  United States  ...        $28            30.0   
2  40.80902 -73.94190  United States  ...       $124             3.0   
3  40.68514 -73.95976  United States  ...        $74            30.0   
4  40.79851 -73.94399  United States  ...        $41            10.0   

  number of reviews last review  reviews per month review rate number  \
0               9.0  10/19/2021               0.21                4.0   
1              45.0   5/21/2022               0.38                4.0   
2               0.0         NaN                NaN                5.0   
3             270.0    7/5/2019               4.64                4.0   
4               9.0  11/19/2018               0.10                3.0   

  calculated host listings count  availability 365  \
0                            6.0             286.0   
1                            2.0             228.0   
2                            1.0             352.0   
3                            1.0             322.0   
4                            1.0             289.0   

                                         house_rules license  
0  Clean up and treat the home the way you'd like...     NaN  
1  Pet friendly but please confirm with me if the...     NaN  
2  I encourage you to use my kitchen, cooking and...     NaN  
3                                                NaN     NaN  
4  Please no smoking in the house, porch or on th...     NaN  

[5 rows x 26 columns]

In [15]:
data.describe()

id       host id            lat           long  \
count  1.025990e+05  1.025990e+05  102591.000000  102591.000000   
mean   2.914623e+07  4.925411e+10      40.728094     -73.949644   
std    1.625751e+07  2.853900e+10       0.055857       0.049521   
min    1.001254e+06  1.236005e+08      40.499790     -74.249840   
25%    1.508581e+07  2.458333e+10      40.688740     -73.982580   
50%    2.913660e+07  4.911774e+10      40.722290     -73.954440   
75%    4.320120e+07  7.399650e+10      40.762760     -73.932350   
max    5.736742e+07  9.876313e+10      40.916970     -73.705220   

       Construction year  minimum nights  number of reviews  \
count      102385.000000   102190.000000      102416.000000   
mean         2012.487464        8.135845          27.483743   
std             5.765556       30.553781          49.508954   
min          2003.000000    -1223.000000           0.000000   
25%          2007.000000        2.000000           1.000000   
50%          2012.000000        3.000000           7.000000   
75%          2017.000000        5.000000          30.000000   
max          2022.000000     5645.000000        1024.000000   

       reviews per month  review rate number  calculated host listings count  \
count       86720.000000       102273.000000                   102280.000000   
mean            1.374022            3.279106                        7.936605   
std             1.746621            1.284657                       32.218780   
min             0.010000            1.000000                        1.000000   
25%             0.220000            2.000000                        1.000000   
50%             0.740000            3.000000                        1.000000   
75%             2.000000            4.000000                        2.000000   
max            90.000000            5.000000                      332.000000   

       availability 365  
count     102151.000000  
mean         141.133254  
std          135.435024  
min          -10.000000  
25%            3.000000  
50%           96.000000  
75%          269.000000  
max         3677.000000

Some columns contain invalid or unrealistic values:

- `minimum nights` has a minimum value of **-1223**, which is not logically possible.
- `availability 365` has negative values.
- Extremely high values in `minimum nights` (up to 5645) suggest data entry errors or outliers.
- The construction year ranges from 2003 to 2022, with a median around 2012.
- This suggests most properties are relatively modern, though this field will require validation.
- The number of reviews shows a highly skewed distribution:Many listings have few or no reviews, A small number of listings have very high review counts



In [ ]:
data.describe(include='object')    

NAME host_identity_verified host name  \
count                102349                 102310    102193   
unique                61281                      2     13190   
top     Home away from home            unconfirmed   Michael   
freq                     33                  51200       881   

       neighbourhood group       neighbourhood        country country code  \
count               102570              102583         102067       102468   
unique                   7                 224              1            1   
top              Manhattan  Bedford-Stuyvesant  United States           US   
freq                 43792                7937         102067       102468   

       instant_bookable cancellation_policy        room type   price  \
count            102494              102523           102599  102352   
unique                2                   3                4    1151   
top               False            moderate  Entire home/apt   $206    
freq              51474               34343            53701     137   

       service fee last review house_rules   license  
count       102326       86706       50468         2  
unique         231        2477        1976         1  
top           $41    6/23/2019      #NAME?  41662/AL  
freq           526        2443        2712         2

- The `price` and `service fee` columns are stored as strings and include currency symbols. These will need to be converted to numeric values during data cleaning.
- `Country` and `country code` has only 1 value and it is of US. 
- Date is also present for the `last review` column but it is not saved as date rather saved as an object
- `instant_bookable` is a boolean column with only 2 values (0, 1)
- `Licence` has only 1 value and almost all of the values are missing from this column


### Next Steps

The identified data quality issues will be addressed in the `data cleaning phase`,
including handling missing values, correcting invalid entries, and treating outliers.categorical variables will be standardized,
missing values handled appropriately, and low-value or redundant columns
considered for removal.
